test/run_pass/test_net_httpclient_parse_error.js

```diff
@@ -39,7 +39,6 @@ request.on('error', function(err) {
 request.end();
 
 process.on('exit', function() {
-  // The first error is a Parse Error.
-  // The second error is the socket hang up.
-  assert.equal(errors, 2);
+  // The error is a Parse Error.
+  assert.equal(errors, 1);
```

Comment:
@daeyeon It seems the API behavior is changed by this PR. Could you explain what is changed?

Reply:
@glistening We roughly throw an `Error` instance with the `socket hang up` message in case that a socket is closed with no response. For instance, that cover cases such as `dns look up error` , `the connection is blocked by server`, and so on. https://github.com/Samsung/iotjs/blob/84600faea6a30352a764c120db8d9bd2965deccd/src/js/http_client.js#L115

However, in the case of `http parsing error`, we have a place throwing the exact error. Thus, in the normal manner, throwing the `Parse Error` only was right. I also checked this test using nodejs, and the result is also `1`, `Parse Error`.

Reply:
@daeyeon Thank you for clear explanation.